### DISCUSSION TITLE
How compatible is minifiaction with CSS in the wild?

### DISCUSSION THREAD

**Discussion by kornelski** [parcel-bundler/lightningcss#667] [General]
Question to project developers: how robust is Lightning CSS? Can I minify any stylesheet from any website, and get a working stylesheet compatible with mainstream browsers (or a reliably reported processing failure)?

I know it has a robust CSS parser, but is the rest of minification similarly robust? Are there any risky/untested features to avoid?

**Comment by devongovett** [comment]
Yes, that is the goal. Lightning CSS is widely used at this point, integrated into multiple bundlers. Early on we tested it with a script that used puppeteer to download the CSS from many popular websites, minify it, and compare the computed styles on all DOM nodes to see if anything was broken. It's been several years since then, and lots of feedback has been incorporated. So I encourage you to try it for your use case and report and if you come across any issues please let us know.

**Comment by kornelski** [comment]
Thanks for the response.

I was about to write such script that checks computed styles on nodes. Is your testing setup available somewhere?

**Reply by devongovett** [reply]
Yeah it's here: https://github.com/parcel-bundler/lightningcss/blob/master/test-integration.mjs

Note that sometimes it can be a bit flaky so if it fails it requires a human to inspect the css and see if it is an actual failure. For example, on websites with transitions or animations, the computed styles may be different between snapshots.
